client/app/components/show-project/actions-list/ulurp-number-link.js

```diff
@@ -0,0 +1,28 @@
+import Component from '@ember/component';
+import ENV from 'labs-zap-search/config/environment';
+
+/**
+ * @param {Action Object} action
+ */
+export default class UlurpNumberLinkComponent extends Component {
+  host = ENV.host;
+
+  ACTION_STATUSCODE_ACTIVE = 1;
+
+  get serverRelativeUrl () {
+    const serverRelativeFolderUrl = this.action.dcpSpabsoluteurl.replace('https://nyco365.sharepoint.com', '');
+    const ulurpNumber = this.action.dcpUlurpnumber;
+
+    // for ulurp numbers that have a letter at the end of the 6 numbers to represent the version (ex. the "A" in C18005AZMX)
+    // pull 6 numbers AND next character after the last number
+    const ulurpNumberWithLetter = (ulurpNumber.match(/(\d+)./g)[0]).toLowerCase();
+    const ulurpNumberWithoutLetter = ulurpNumber.match(/\d+/g)[0]; // pull 100149 from C100149ZSM
+
+    // the ulurp numbers with that extra letter have overall 11 characters compared to the usual 10
+    if (ulurpNumber.length > 10) {
+      return `${serverRelativeFolderUrl}/${ulurpNumberWithLetter}.pdf`;
+    }
+
+    return `${serverRelativeFolderUrl}/${ulurpNumberWithoutLetter}.pdf`;
```

Comment:
This was hard for me to grock but understand it's a weird formatting. I don't know how wrote this originally, and I realize you're moving code over from something else, but I would consider ways to improve this code.

Reply:
@allthesignals  I've refactored this to make the code a little closer to human-language, I think. Take a look and lmk if you think so!

Reply:
I've refactored, hoping it'll make it clearer!

Reply:
Great!